api4/channel.go

```diff
@@ -690,8 +690,8 @@ func getPinnedPosts(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getAllChannels(c *Context, w http.ResponseWriter, r *http.Request) {
-	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
-		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_GROUPS) || !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_CHANNELS) {
+		c.SetPermissionsError([]*model.Permission{model.PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_GROUPS, model.PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_CHANNELS})
```

Comment:
TODO: For all `PERMISSION_*_SYSCONSOLE_USERMANAGEMENT_*` permission ensure there's an OR condition on the `PERMISSION_*_SYSCONSOLE_USERMANAGEMENT` so that system admin doesn't require all of the sub permissions and that the main permission covers all of the sub-permissions.